tgui/packages/tgui/interfaces/common/SortButton.js

```diff
@@ -0,0 +1,33 @@
+import { useLocalState } from '../../backend';
+import {
+  Button,
+  Icon,
+  Table,
+} from '../../components';
+
+export const SortButton = (properties, context) => {
```

Comment:
This being abstracted out worries me. Even though you're cutting out duplicate code, you're also essentially creating a component with a very niche usage that also comes with strings attached to it (i.e. you can't just use it out of the box) unlike most other common JSX elements.

I say this because future contributors won't know that this uses `LocalState` in a way that makes it difficult to use more than 1 of these elements in any given interface for reasons you've already seen.